recipes/cpr/all/conanfile.py

```diff
@@ -34,38 +55,58 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
+        if not self._supports_openssl:
+            del self.options.with_openssl
+        if not self._supports_winssl:
+            del self.options.with_winssl
+
+        # Make sure libcurl uses the same SSL implementation
+        if self.options.get_safe("with_openssl", False):
+            # self.options["libcurl"].with_openssl = True # deprecated in https://github.com/conan-io/conan-center-index/pull/2880
+            self.options["libcurl"].with_ssl = "openssl"
+        if self.options.get_safe("with_winssl", False):
+            # self.options["libcurl"].with_winssl = True # deprecated in https://github.com/conan-io/conan-center-index/pull/2880
+            self.options["libcurl"].with_ssl = "schannel"
```

Comment:
How about, instead of introducing duplicate options, deprecate the ssl related options of cpr and just use the options of libcurl?
So use `self.options["libcurl"].with_ssl == "openssl"` instead of `self.options.with_openssl`.

Reply:
Because depending on the versions available is depending on the supported values... that was my initial logic.

cpr does not claim to support many ssl layers, I am not familiar with it enough know it it can be matched to libcurl

My problem is very much related

https://github.com/conan-io/conan-center-index/blob/2e6694cf675476fcaa46e53d77d46f859c080750/recipes/libcurl/all/conanfile.py#L95

Reply:
Indeed, the default value on macos of libcurl is different.

And by leaning on the libcurl options, you avoid duplicating this logic.